recipes/sqlite3mc/all/conanfile.py

```diff
@@ -84,7 +84,7 @@ class sqlite3mc(ConanFile):
         "enable_rtree": True,
         "enable_uuid": True,
         "use_uri": True,
-        "user_authentication": True,
+        "user_authentication": False,
```

Comment:
I see it was change in 1.8.4, is that correct? https://github.com/utelle/SQLite3MultipleCiphers/commit/3bb033956816b3301f026abb5e83087799de5bee

I would suggest updating the `config_options` to follow the default behavior to honor older versions too:

```
def config_options(self):
    if Version(self.version) < "1.8.4":
        # INFO: https://github.com/utelle/SQLite3MultipleCiphers/commit/3bb033956816b3301f026abb5e83087799de5bee
        self.options.user_authentication = True
```

Reply:
> I see it was change in 1.8.4, is that correct? [utelle/SQLite3MultipleCiphers@3bb0339](https://github.com/utelle/SQLite3MultipleCiphers/commit/3bb033956816b3301f026abb5e83087799de5bee)

Yes, the _user authentication extension_ was deprecated by the _SQLite core developers_. It may disappear with the next release of _SQLite_ (or anytime later). Therefore I decided to disable it by default.

Whether I will remove the code base of the extension from my project will depend on how the _SQLite team_ will handle this extension in the future.

> I would suggest updating the `config_options` to follow the default behavior to honor older versions too:
> 
> ```
> def config_options(self):
>     if Version(self.version) < "1.8.4":
>         # INFO: https://github.com/utelle/SQLite3MultipleCiphers/commit/3bb033956816b3301f026abb5e83087799de5bee
>         self.options.user_authentication = True
> ```

I can do that. However, I think I will wait with updating the Conan recipe, because I just saw that the next SQLite version will probably be released within the next 2 weeks.

Reply:
Thank you for giving more details. We could deprecate that option as well instead: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#can-i-remove-an-option-from-a-recipe

Than few versions later we can remove from package (if the project does the same).